CMakeLists.txt

```diff
@@ -19,6 +19,9 @@ option(debug    "Compile with debug flags"                       OFF)
 option(optimize "Turn on all compiler optimization flags"        OFF)
 option(coverage "Compile with coverage analysis flags"           OFF)
 option(mpif08   "Use Fortran 2008 MPI interface"                 OFF)
+option(dagmc    "Enable support for DAGMC (CAD) geometry"        OFF)
```

Comment:
Let's add a check below that throws an error if a user has dagmc=on and openmp=on (until the MOAB thread safety issues are resolved).

Reply:
Maybe a warning? Threads still work with CSG geometry and users may want to run either geom type with the same executable. 

I can force num_threads = 1 if it's a dagmc run and give a message to the user that this is happening.

Reply:
Ah, good point. In that case a warning is probably fine.

Reply:
Ok I'll update that. Can't wait to remove it later :)